algorithms/symmetry/cosym/__init__.py

```diff
@@ -380,14 +380,18 @@ def _reindexing_ops_for_dataset(self, dataset_id, sym_ops, cosets):
         )
 
         for i_cluster in range(n_clusters):
-            isel = (self.cluster_labels == i_cluster).iselection()
-            dataset_ids = isel % len(self.input_intensities)
-            sel = (dataset_ids == dataset_id).iselection()
-            for s in sel:
-                sym_op_id = isel[s] // len(self.input_intensities)
-                for partition in cosets.partitions:
-                    if sym_ops[sym_op_id] in partition:
-                        if i_cluster not in reindexing_ops:
+            # Select all points within this cluster
```

Comment:
This is quite a large re-arrangement of code.... which looks like something which would benefit from a unit test to show that this is tidying up / refactoring not changing behaviour (and if that is a bug fix, adding a regession test)

Reply:
It's mostly just renaming variables and adding comments for clarity (added as I re-looked at the code to figure out what is going on). The only real change is the addition of the `break` and moving `if i_cluster not in reindexing_ops` to the middle loop. Previously it would have continued iterating over the inner two loops after adding `i_cluster` to `reindexing_ops`, but not doing anything because of the inner `if i_cluster not in reindexing_ops` check. Now with the addition of the `break` and moving the `if` statement up a level, it will effectively skip straight to the next outer loop.

Reply:
Looking at this again, I wonder if changing this to:
```
                if i_cluster in reindexing_ops:
                    break
```
and dedenting the rest of the code might be better.

Reply:
That may well be clearer. 

I am not questioning the correctness of the code, only how easy it is for a person-in-the-future to understand

Reply:
Added a minimal test in https://github.com/dials/dials/pull/1514/commits/0ed77e801985d37175e5fe13be7fc795d1adf426. I cherry-picked this commit to master and it passed there too.